// simple-regex/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Provides a way to test if a number is in one of the valid mobile number patterns
 */
const TestMobileNumber = (mobileNumber) => {
    const isValidMobileNumber = exports.MobileNumber.Econet.test(mobileNumber) || exports.MobileNumber.Telecel.test(mobileNumber) || exports.MobileNumber.Netone.test(mobileNumber);
    return isValidMobileNumber;
};
exports.EmailAddress = RegExp('^[a-zA-Z0-9.!#$%&’+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:.[a-zA-Z0-9-]+)$');
exports.MobileNumber = {
    Econet: RegExp(/^(((\+)*263)|0)7(7|8)[0-9]{7}$/),
    Netone: RegExp(/^(((\+)*263)|0)71[0-9]{7}$/),
    Telecel: RegExp(/^(((\+)*263)|0)73[0-9]{7}$/),
    test: TestMobileNumber
};
exports.NationalID = RegExp(/^[0-9]{2}(-|s)[0-9]{6,7}s*[A-Z]s*[0-9]{2}$/);
exports.NumberPlate = RegExp(/^[A-Z]{3}(-|[[:blank:]])[0-9]{3}$/gm);
exports.PassportNumber = RegExp(/^[A-Z]{2}[0-9]{6}$/);
exports.DriversLicence = RegExp(/^\b\d{5}[a-zA-Z]{2}\b/);
exports.SSN = RegExp(/^\b([0-9]){7}([a-zA-Z]){1}\b/);
exports.SimpleRegex = {
    /**
     * Provides regular expression for Drivers' Licences in Zimbabwe
     */
    DriversLicence: exports.DriversLicence,
    /**
     * Provides regular expression for email addresses
     */
    EmailAddress: exports.EmailAddress,
    /**
     * Regular expressions for mobile phone network operators in Zimbabwe.
     */
    MobileNumber: exports.MobileNumber,
    /**
     * Regular expression for National Identity Numbers in Zimbabwe
     */
    NationalID: exports.NationalID,
    /**
     * Regular expression for ordinary motor vehicle licence plate numbers in Zimbabwe.
     */
    NumberPlate: exports.NumberPlate,
    /**
     * Regular expression passport numbers in Zimbabwe.
     */
    PassportNumber: exports.PassportNumber,
    /**
     * Regular expression for social security number by National Social Security Authority of Zimbabwe.
     */
    SSN: exports.SSN,
};
